runtime/src/chpl-visual-debug.c

```diff
@@ -224,58 +245,90 @@ void chpl_vdebug_pause (void) {
+// Record>  nb_get: time.sec dstNodeId srcNodeId commTaskId addr raddr elemsize 
+//                  typeIndex length lineNumber fileName
+//
+// Note: dstNodeId is node requesting Get
+
 void chpl_vdebug_log_get_nb(void* addr, c_nodeid_t node, void* raddr,
                             int32_t elemSize, int32_t typeIndex,
                             int32_t len, int ln, c_string fn) {
   if (chpl_vdebug) {
     struct timeval tv;
     struct timezone tz = {0,0};
+    chpl_taskID_t commTask = chpl_task_getId();
     (void) gettimeofday (&tv, &tz);
-    chpl_dprintf (chpl_vdebug_fd, "nb_get: %lld.%06ld %d %d 0x%lx 0x%lx %d %d %d %d %s\n",
+    chpl_dprintf (chpl_vdebug_fd, "nb_get: %lld.%06ld %d %d %lu 0x%lx 0x%lx %d %d %d %d %s\n",
                   (long long) tv.tv_sec, (long) tv.tv_usec,  chpl_nodeID, node,
-                  (long) addr, (long) raddr, elemSize, typeIndex, len, ln, fn);
+                  (unsigned long)commTask, (long) addr, (long) raddr, elemSize, typeIndex, len,
+                  ln, fn);
   }
 }
 
+// Record>  put: time.sec srcNodeId dstNodeId commTaskId addr raddr elemsize 
+//               typeIndex length lineNumber fileName
+
+
 void chpl_vdebug_log_put(void* addr, c_nodeid_t node, void* raddr,
                          int32_t elemSize, int32_t typeIndex, int32_t len,
                          int ln, c_string fn) {
   if (chpl_vdebug) {
     struct timeval tv;
     struct timezone tz = {0,0};
+    chpl_taskID_t commTask = chpl_task_getId();
     (void) gettimeofday (&tv, &tz);
-    chpl_dprintf (chpl_vdebug_fd, "put: %lld.%06ld %d %d 0x%lx 0x%lx %d %d %d %d %s\n",
-                  (long long) tv.tv_sec, (long) tv.tv_usec,  chpl_nodeID, node,
-                  (long) addr, (long) raddr, elemSize, typeIndex, len, ln, fn);
+    chpl_dprintf (chpl_vdebug_fd, "put: %lld.%06ld %d %d %lu 0x%lx 0x%lx %d %d %d %d %s\n",
+                  (long long) tv.tv_sec, (long) tv.tv_usec, chpl_nodeID, node,
+                  (unsigned long) commTask, (long) addr, (long) raddr, elemSize, typeIndex, len,
+                  ln, fn);
   }
 }
 
+// Record>  get: time.sec dstNodeId srcNodeId commTaskId addr raddr elemsize 
+//               typeIndex length lineNumber fileName
+//
+// Note:  dstNodeId is for the node makeing the request
+
 void chpl_vdebug_log_get(void* addr, c_nodeid_t node, void* raddr,
                          int32_t elemSize, int32_t typeIndex, int32_t len,
                          int ln, c_string fn) {
-  if (chpl_vdebug && (chpl_task_getId() != chpl_vdebug_task)) {
+  if (chpl_vdebug) {
     struct timeval tv;
     struct timezone tz = {0,0};
+    chpl_taskID_t commTask = chpl_task_getId();
     (void) gettimeofday (&tv, &tz);
-    chpl_dprintf (chpl_vdebug_fd, "get: %lld.%06ld %d %d 0x%lx 0x%lx %d %d %d %d %s\n",
-                  (long long) tv.tv_sec, (long) tv.tv_usec,  chpl_nodeID, node,
-                  (long) addr, (long) raddr, elemSize, typeIndex, len, ln, fn);
+    /// XXXX bug I can't fix!!!
```

Comment:
Okay. If you're planning on merging the PR before tracking down the problem and figuring out a way to deal with it, then please put a description of the problem in place of the "bug" comment. (Just in case something unforeseen prevents you from getting back to it.)